### PR TITLE
NIFI-12263: Upgrading AWS Machine Learning processors to SDK 2.x

### PR DESCRIPTION
**Pull request by gresockj**
See the Machine Learning flow definition attached to the JIRA for testing.

# Summary

[NIFI-12263](https://issues.apache.org/jira/browse/NIFI-12263)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [ ] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [ ] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [ ] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [ ] Pull Request based on current revision of the `main` branch
- [ ] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [ ] Build completed using `mvn clean install -P contrib-check`
  - [ ] JDK 21

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-nar-bundles/nifi-aws-bundle/nifi-aws-abstract-processors/pom.xml [hunk 1]
```diff
@@ -130,6 +146,11 @@
             <groupId>com.github.ben-manes.caffeine</groupId>
             <artifactId>caffeine</artifactId>
         </dependency>
+        <dependency>
+            <groupId>com.fasterxml.jackson.datatype</groupId>
+            <artifactId>jackson-datatype-jsr310</artifactId>
+            <version>2.15.3</version>
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-aws-bundle/nifi-aws-abstract-processors/pom.xml:152]: This version should not be set explicitly. It should be provided in the `jackson-bom` at the root Maven configuration, did you notice any issues with the version not set?

**Review by exceptionfactory — COMMENTED** [review]
Thanks for working on this migration @gresockj, I noted one question about the Jackson library dependency version.

File: nifi-nar-bundles/nifi-aws-bundle/nifi-aws-abstract-processors/pom.xml (hunk 1 above)
Comment (gresockj) [reply nifi-nar-bundles/nifi-aws-bundle/nifi-aws-abstract-processors/pom.xml]: No, I can remove the explicit version here.

**Review by exceptionfactory — APPROVED** [review]
Thanks for making the adjustment @gresockj, the latest version looks good! +1 merging

**Pull request closed without merging**